pkg/provisioner/ironic/ironic.go

```diff
@@ -234,6 +234,14 @@ func (p *ironicProvisioner) ValidateManagementAccess(credentialsChanged bool) (r
 	if ironicNode == nil {
 		p.log.Info("registering host in ironic")
 
+		properties := map[string]interface{}{}
+
+		if p.host.Spec.BootMode == metal3v1alpha1.Legacy {
+			properties["capabilities"] = "boot_mode:bios"
+		} else if p.host.Spec.BootMode == metal3v1alpha1.UEFI {
+			properties["capabilities"] = "boot_mode:uefi"
+		}
```

Comment:
Is there any reason not to add an "else" case here to cause us to pass an explicit value to Ironic to control its behavior?

Reply:
If we set it , it should need to be "bios" to don't disrupt the current default behaviour. But on the other side we are more interested in booting from UEFI for several features (redfish, ipv6...) So really what could be a sane default there? According to discussion, i opted to just don't enforce any value, but don't mind revisiting it.

Reply:
We should be explicit and not rely on Ironic's default behavior. I agree we should set the default to the current value.

Reply:
Explicitly passing UEFI or BIOS when the user hasn't specified makes sense only for a server that supports both UEFI and BIOS. In practice this means x86 servers released in a particular time period. It doesn't make sense (and IIUC will fail) to pass something when the server supports only the other one or neither.
Our options as I see it are to either:
* Ask Ironic which ones are supported, choose one from a predefined priority list, and pass that explicitly. Document the priority list.
* Don't pass anything. Document what Ironic does by default as the default.

Reply:
First option adds extra complexity and I am not sure that we have this capability to query ironic for that.
The "do not pass anything" approach was considered, by just adding documentation on the default being BIOS if not set. But this is conflicting with other reviews. @dhellmann  @sadasu , what are your thoughts there?

Reply:
When we support platforms other than x86 for bare metal, then I agree we would need to change the behavior and document the appropriate default on each platform. But I would prefer to have the metal3 default behavior be deterministic and discoverable by looking at metal3 code, which it isn't if we let Ironic decide what the default behavior is.

Reply:
@dhellmann @yrobla I agree that we should keep the implementation here (baremetal-operator) independent of the defaults in Ironic or what the hw is capable of for that matter. 
If bios or uefi is explicitly specified in the CR, then we are doing the right thing.
If the boot mode field is not set in the CR (version skew during upgrades and downgrades + other scenarios), then we currently fall through to Ironic's way of handling it. That was my concern during the previous round. If we want to always set a value explicitly then we need to set one when the boot mode is not set in the CR.

The 3rd consideration is what if the value set by baremetal-operator is not supported in Ironic + HW combo. To cover that case, we need to revise documentation in the CRD to say something like : baremetal-operator supports specific default X. If your HW supports RedFish, please set the bootmode here to UEFI. (The user should be aware of what teir HW is capable of even if they don't know about Ironic.) If the boot mode set here is not supported on the HW or unsupported in our underlying service (Ironic), then the actual/observed boot mode might differ from value set in CRD. Add link to Ironic documentation here to show how actual boot mode is dependent of type of HW, ability to read the value from HW etc.

Reply:
> The 3rd consideration is what if the value set by baremetal-operator is not supported in Ironic + HW combo. To cover that case, we need to revise documentation in the CRD to say something like : baremetal-operator supports specific default X. If your HW supports RedFish, please set the bootmode here to UEFI. (The user should be aware of what teir HW is capable of even if they don't know about Ironic.) If the boot mode set here is not supported on the HW or unsupported in our underlying service (Ironic), then the actual/observed boot mode might differ from value set in CRD. Add link to Ironic documentation here to show how actual boot mode is dependent of type of HW, ability to read the value from HW etc.

We should not ever under any circumstances force metal3 users to read Ironic's documentation to understand what behavior they will get. Ironic should be completely hidden from them.

If we can determine the right boot mode based on other inputs (e.g., the use of Redfish, or virtual media, or IPv6), then we should use those indicators instead of adding another field that the user has to set correctly for themselves.

I'm starting to wonder if we need the field at all, or if we can just default to UEFI and come back and address this again when we have support for other hardware types.